app/src/main/java/com/bogdan/codeforceswatcher/features/users/AddUserBottomSheetFragment.kt

```diff
@@ -0,0 +1,127 @@
+package com.bogdan.codeforceswatcher.features.users
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.LinearProgressIndicator
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.components.compose.buttons.SmallButton
+import com.bogdan.codeforceswatcher.components.compose.textfields.HandleTextField
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import io.xorum.codeforceswatcher.features.users.redux.UsersRequests
+import io.xorum.codeforceswatcher.features.users.redux.UsersState
+import io.xorum.codeforceswatcher.redux.store
+import tw.geothings.rekotlin.StoreSubscriber
+
+class AddUserBottomSheetFragment: BottomSheetDialogFragment(), StoreSubscriber<UsersState> {
+
+    private val addUserStatus: MutableState<UsersState.Status?> = mutableStateOf(null)
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): ComposeView {
+        dialog?.window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
+
+        return ComposeView(requireContext()).apply {
+            setContent {
+                AlgoismeTheme {
+                    ContentView(addUserStatus) {
+                        store.dispatch(UsersRequests.AddUser(it))
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        store.subscribe(this) { state ->
+            state.skipRepeats { oldState, newState ->
+                oldState.users == newState.users
+            }.select { it.users }
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+
+        store.unsubscribe(this)
+    }
+
+    override fun onNewState(state: UsersState) {
+        addUserStatus.value = state.addUserStatus
+
+        if (addUserStatus.value == UsersState.Status.DONE) {
+            dismiss()
+        }
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun ContentView(
```

Comment:
We usually try to name the top-level view as the activity/fragment. In this case, it would be `AddUserView`.

Reply:
But you said we can name it just ContentView🤔😅

Reply:
I've probably missed the point :) For the most top-level composables it usually reflects the activity/fragment, but for all lower-level composables names can be simpler (`ContentView` is usually used for 2nd level composable responsible for content of the screen, on the same level we often have `LoadingView`, `ErrorView`, etc.)